session.go

```diff
@@ -296,6 +301,12 @@ func ParseURL(url string) (*DialInfo, error) {
 			if err != nil {
 				return nil, errors.New("bad value for maxPoolSize: " + opt.value)
 			}
+		case "appname":
+			appName = opt.value
+			// make sur sizeOf(appName) < 128 bytes
+			if len(appName) > 32 {
```

Comment:
This can be bumped to `len(appName) > 128` - I double checked and the BSON encoded metadata doesn't exceed 512 bytes with this change.

Reply:
I'd appreciate if we error out instead of truncating the metadata. Or at least document the maximum len properly. Additionally you can truncate the `appName` in a middle of an multi byte utf-8 rune, which will be ugly.